python/tests/test_tools/test_toolset.py

```diff
@@ -328,3 +329,64 @@ def test_execute_action_param_serialization() -> None:
         text=None,
         session_id=mock.ANY,
     )
+
+
+class TestSubclassInit:
+
+    def test_runtime(self):
+
+        class SomeToolsetExtention(ComposioToolSet):
+            pass
+
+        assert (
+            SomeToolsetExtention._runtime  # pylint: disable=protected-access
+            == "composio"
+        )
+
+        class SomeOtherToolsetExtention(ComposioToolSet, runtime="some_toolset"):
+            pass
+
+        assert (
+            SomeOtherToolsetExtention._runtime  # pylint: disable=protected-access
+            == "some_toolset"
+        )
+
+    def test_description_char_limit(self) -> None:
+
+        char_limit = 512
+        (schema,) = ComposioToolSet().get_action_schemas(
+            actions=[
+                Action.FILETOOL_GIT_CLONE,
+            ]
+        )
+        assert len(t.cast(str, schema.description)) > char_limit
+
+        class SomeToolsetExtention(ComposioToolSet, description_char_limit=char_limit):
+            pass
+
+        (schema,) = SomeToolsetExtention().get_action_schemas(
+            actions=[
+                Action.FILETOOL_GIT_CLONE,
+            ]
+        )
+        assert len(t.cast(str, schema.description)) == char_limit
+
+    def test_action_name_char_limit(self) -> None:
```

Comment:
Consider adding test cases for edge cases:
1. Test with limit=0 (should preserve empty string)
2. Test with limit=1 (minimal valid length)
3. Test with negative limit (should raise ValueError)
4. Test with very large limit (should preserve full name)

This would help ensure robust handling of all possible input values.